[2024-03-05T21:13:44.933+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-05T21:13:44.937+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-05T21:13:44.937+0000] {taskinstance.py:2170} INFO - Starting attempt 4 of 4
[2024-03-05T21:13:44.944+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): download_dataset> on 2022-12-01 00:00:00+00:00
[2024-03-05T21:13:44.948+0000] {standard_task_runner.py:60} INFO - Started process 188 to run task
[2024-03-05T21:13:44.949+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'citibikedataset', 'download_dataset', 'scheduled__2022-12-01T00:00:00+00:00', '--job-id', '68', '--raw', '--subdir', 'DAGS_FOLDER/dag_citibikenyc.py', '--cfg-path', '/tmp/tmpyozawa0y']
[2024-03-05T21:13:44.950+0000] {standard_task_runner.py:88} INFO - Job 68: Subtask download_dataset
[2024-03-05T21:13:44.975+0000] {task_command.py:423} INFO - Running <TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [running]> on host 8277132cfd53
[2024-03-05T21:13:44.998+0000] {abstractoperator.py:707} ERROR - Exception rendering Jinja template for task 'download_dataset', field 'bash_command'. Template: "sh ${AIRFLOW_HOME}/include/download_bike_dataset.sh {{ ds.strftime('%Y') }} {{ ds.strftime('%m') }}"
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 699, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 174, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 654, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 131, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/sandbox.py", line 391, in call
    if not __self.is_safe_callable(__obj):
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/sandbox.py", line 275, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 859, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 852, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'str object' has no attribute 'strftime'
[2024-03-05T21:13:44.999+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2334, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode, session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2465, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2877, in render_templates
    original_task.render_template_fields(context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 1241, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 699, in _do_render_template_fields
    rendered_content = self.render_template(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 174, in render_template
    return self._render(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/abstractoperator.py", line 654, in _render
    return super()._render(template, context, dag=dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/template/templater.py", line 131, in _render
    return render_template_to_string(template, context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/sandbox.py", line 391, in call
    if not __self.is_safe_callable(__obj):
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/sandbox.py", line 275, in is_safe_callable
    getattr(obj, "unsafe_callable", False) or getattr(obj, "alters_data", False)
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 859, in __getattr__
    return self._fail_with_undefined_error()
  File "/home/airflow/.local/lib/python3.8/site-packages/jinja2/runtime.py", line 852, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'str object' has no attribute 'strftime'
[2024-03-05T21:13:45.002+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=citibikedataset, task_id=download_dataset, execution_date=20221201T000000, start_date=20240305T211344, end_date=20240305T211345
[2024-03-05T21:13:45.008+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 68 for task download_dataset ('str object' has no attribute 'strftime'; 188)
[2024-03-05T21:13:45.041+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-03-05T21:13:45.060+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-03-07T13:47:05.431+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-07T13:47:05.435+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-07T13:47:05.436+0000] {taskinstance.py:2170} INFO - Starting attempt 4 of 5
[2024-03-07T13:47:05.443+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): download_dataset> on 2022-12-01 00:00:00+00:00
[2024-03-07T13:47:05.446+0000] {standard_task_runner.py:60} INFO - Started process 601 to run task
[2024-03-07T13:47:05.447+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'citibikedataset', 'download_dataset', 'scheduled__2022-12-01T00:00:00+00:00', '--job-id', '130', '--raw', '--subdir', 'DAGS_FOLDER/dag_citibikenyc.py', '--cfg-path', '/tmp/tmp0a301cjc']
[2024-03-07T13:47:05.448+0000] {standard_task_runner.py:88} INFO - Job 130: Subtask download_dataset
[2024-03-07T13:47:05.473+0000] {task_command.py:423} INFO - Running <TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [running]> on host 0430cbdaac75
[2024-03-07T13:47:05.500+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
[2024-03-07T13:47:05.614+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='felipeparfitt' AIRFLOW_CTX_DAG_ID='citibikedataset' AIRFLOW_CTX_TASK_ID='download_dataset' AIRFLOW_CTX_EXECUTION_DATE='2022-12-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-01T00:00:00+00:00'
[2024-03-07T13:47:05.615+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-03-07T13:47:05.616+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'sh ${AIRFLOW_HOME}/include/download_bike_dataset.sh         2023 01']
[2024-03-07T13:47:05.622+0000] {subprocess.py:86} INFO - Output:
[2024-03-07T13:47:05.623+0000] {subprocess.py:93} INFO - 
[2024-03-07T13:47:05.623+0000] {subprocess.py:93} INFO - Downloading citibike dataset from 2023/01
[2024-03-07T13:47:05.628+0000] {subprocess.py:93} INFO -   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[2024-03-07T13:47:05.628+0000] {subprocess.py:93} INFO -                                  Dload  Upload   Total   Spent    Left  Speed
[2024-03-07T13:47:12.344+0000] {subprocess.py:93} INFO -   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0 1820k    0 17004    0     0  17480      0  0:01:46 --:--:--  0:01:46 17475 10 1820k   10  186k    0     0   104k      0  0:00:17  0:00:01  0:00:16  104k 27 1820k   27  492k    0     0   176k      0  0:00:10  0:00:02  0:00:08  176k 44 1820k   44  814k    0     0   215k      0  0:00:08  0:00:03  0:00:05  215k 62 1820k   62 1138k    0     0   238k      0  0:00:07  0:00:04  0:00:03  238k 82 1820k   82 1495k    0     0   258k      0  0:00:07  0:00:05  0:00:02  307k100 1820k  100 1820k    0     0   271k      0  0:00:06  0:00:06 --:--:--  331k
[2024-03-07T13:47:12.348+0000] {subprocess.py:93} INFO - Archive:  /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv.zip
[2024-03-07T13:47:12.391+0000] {subprocess.py:93} INFO -   inflating: /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv
[2024-03-07T13:47:12.391+0000] {subprocess.py:93} INFO -   inflating: /opt/***/include/dataset/raw/2023/01/__MACOSX/._JC-202301-citibike-tripdata.csv
[2024-03-07T13:47:12.563+0000] {subprocess.py:93} INFO -   adding: opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv (deflated 84%)
[2024-03-07T13:47:12.564+0000] {subprocess.py:93} INFO - LOCAL_PATH=/opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv.zip
[2024-03-07T13:47:12.564+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2024-03-07T13:47:12.577+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=citibikedataset, task_id=download_dataset, execution_date=20221201T000000, start_date=20240307T134705, end_date=20240307T134712
[2024-03-07T13:47:12.589+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2024-03-07T13:47:12.598+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-03-07T17:49:34.914+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-07T17:49:34.919+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-07T17:49:34.919+0000] {taskinstance.py:2170} INFO - Starting attempt 4 of 5
[2024-03-07T17:49:34.927+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): download_dataset> on 2022-12-01 00:00:00+00:00
[2024-03-07T17:49:34.931+0000] {standard_task_runner.py:60} INFO - Started process 213 to run task
[2024-03-07T17:49:34.934+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'citibikedataset', 'download_dataset', 'scheduled__2022-12-01T00:00:00+00:00', '--job-id', '202', '--raw', '--subdir', 'DAGS_FOLDER/dag_citibikenyc.py', '--cfg-path', '/tmp/tmpgpyf4vim']
[2024-03-07T17:49:34.935+0000] {standard_task_runner.py:88} INFO - Job 202: Subtask download_dataset
[2024-03-07T17:49:34.965+0000] {task_command.py:423} INFO - Running <TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [running]> on host a0c898e5643f
[2024-03-07T17:49:35.003+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
[2024-03-07T17:49:35.024+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='felipeparfitt' AIRFLOW_CTX_DAG_ID='citibikedataset' AIRFLOW_CTX_TASK_ID='download_dataset' AIRFLOW_CTX_EXECUTION_DATE='2022-12-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-01T00:00:00+00:00'
[2024-03-07T17:49:35.025+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-03-07T17:49:35.025+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'sh ${AIRFLOW_HOME}/include/download_bike_dataset.sh         2023 01']
[2024-03-07T17:49:35.032+0000] {subprocess.py:86} INFO - Output:
[2024-03-07T17:49:35.035+0000] {subprocess.py:93} INFO - 
[2024-03-07T17:49:35.035+0000] {subprocess.py:93} INFO - Downloading citibike dataset from 2023/01
[2024-03-07T17:49:35.040+0000] {subprocess.py:93} INFO -   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[2024-03-07T17:49:35.040+0000] {subprocess.py:93} INFO -                                  Dload  Upload   Total   Spent    Left  Speed
[2024-03-07T17:49:37.751+0000] {subprocess.py:93} INFO -   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0 21 1820k   21  390k    0     0   169k      0  0:00:10  0:00:02  0:00:08  169k100 1820k  100 1820k    0     0   671k      0  0:00:02  0:00:02 --:--:--  671k
[2024-03-07T17:49:37.754+0000] {subprocess.py:93} INFO - Archive:  /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv.zip
[2024-03-07T17:49:37.795+0000] {subprocess.py:93} INFO -   inflating: /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv
[2024-03-07T17:49:37.796+0000] {subprocess.py:93} INFO -   inflating: /opt/***/include/dataset/raw/2023/01/__MACOSX/._JC-202301-citibike-tripdata.csv
[2024-03-07T17:49:37.971+0000] {subprocess.py:93} INFO -   adding: opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv (deflated 84%)
[2024-03-07T17:49:37.972+0000] {subprocess.py:93} INFO - 
[2024-03-07T17:49:37.972+0000] {subprocess.py:93} INFO - Download completed successfully!
[2024-03-07T17:49:37.972+0000] {subprocess.py:93} INFO - /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv.zip
[2024-03-07T17:49:37.972+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2024-03-07T17:49:37.985+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=citibikedataset, task_id=download_dataset, execution_date=20221201T000000, start_date=20240307T174934, end_date=20240307T174937
[2024-03-07T17:49:37.996+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2024-03-07T17:49:38.010+0000] {taskinstance.py:3280} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-03-07T20:25:06.064+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-07T20:25:06.164+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [queued]>
[2024-03-07T20:25:06.165+0000] {taskinstance.py:2170} INFO - Starting attempt 4 of 5
[2024-03-07T20:25:06.172+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): download_dataset> on 2022-12-01 00:00:00+00:00
[2024-03-07T20:25:06.176+0000] {standard_task_runner.py:60} INFO - Started process 4460 to run task
[2024-03-07T20:25:06.178+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'citibikedataset', 'download_dataset', 'scheduled__2022-12-01T00:00:00+00:00', '--job-id', '315', '--raw', '--subdir', 'DAGS_FOLDER/dag_citibikenyc.py', '--cfg-path', '/tmp/tmpl430y4ns']
[2024-03-07T20:25:06.179+0000] {standard_task_runner.py:88} INFO - Job 315: Subtask download_dataset
[2024-03-07T20:25:06.205+0000] {task_command.py:423} INFO - Running <TaskInstance: citibikedataset.download_dataset scheduled__2022-12-01T00:00:00+00:00 [running]> on host a0c898e5643f
[2024-03-07T20:25:06.231+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
[2024-03-07T20:25:06.247+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='felipeparfitt' AIRFLOW_CTX_DAG_ID='citibikedataset' AIRFLOW_CTX_TASK_ID='download_dataset' AIRFLOW_CTX_EXECUTION_DATE='2022-12-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-01T00:00:00+00:00'
[2024-03-07T20:25:06.249+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-03-07T20:25:06.249+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'sh ${AIRFLOW_HOME}/include/download_bike_dataset.sh         2023 01']
[2024-03-07T20:25:06.257+0000] {subprocess.py:86} INFO - Output:
[2024-03-07T20:25:06.261+0000] {subprocess.py:93} INFO - 
[2024-03-07T20:25:06.261+0000] {subprocess.py:93} INFO - Downloading citibike dataset from 2023/01
[2024-03-07T20:25:06.267+0000] {subprocess.py:93} INFO -   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[2024-03-07T20:25:06.267+0000] {subprocess.py:93} INFO -                                  Dload  Upload   Total   Spent    Left  Speed
[2024-03-07T20:25:07.966+0000] {subprocess.py:93} INFO -   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0 1820k    0 17004    0     0  19281      0  0:01:36 --:--:--  0:01:36 19278100 1820k  100 1820k    0     0  1071k      0  0:00:01  0:00:01 --:--:-- 1071k
[2024-03-07T20:25:07.971+0000] {subprocess.py:93} INFO - Archive:  /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv.zip
[2024-03-07T20:25:08.013+0000] {subprocess.py:93} INFO -   inflating: /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv
[2024-03-07T20:25:08.013+0000] {subprocess.py:93} INFO -   inflating: /opt/***/include/dataset/raw/2023/01/__MACOSX/._JC-202301-citibike-tripdata.csv
[2024-03-07T20:25:08.188+0000] {subprocess.py:93} INFO -   adding: opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv (deflated 84%)
[2024-03-07T20:25:08.190+0000] {subprocess.py:93} INFO - 
[2024-03-07T20:25:08.190+0000] {subprocess.py:93} INFO - Download completed successfully!
[2024-03-07T20:25:08.190+0000] {subprocess.py:93} INFO - /opt/***/include/dataset/raw/2023/01/JC-202301-citibike-tripdata.csv.zip
[2024-03-07T20:25:08.190+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2024-03-07T20:25:08.211+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=citibikedataset, task_id=download_dataset, execution_date=20221201T000000, start_date=20240307T202506, end_date=20240307T202508
[2024-03-07T20:25:08.237+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2024-03-07T20:25:08.257+0000] {taskinstance.py:3280} INFO - 1 downstream tasks scheduled from follow-on schedule check
